views/actions/packages.js

```diff
@@ -79,25 +82,61 @@ export function resetGetPackagesError() {
   };
 }
 
+export function addPackageHistory(history) {
+  return {
+    type: ADD_PACKAGE_HISTORY,
+    history,
+  };
+}
+
 function createGetPackagesRequest() {
   return {
     path: '/packages',
     method: 'GET',
   };
 }
 
+function createGetPackageHistoryRequest(id) {
+  return {
+    path: `/packages/${id}/packageHistory`,
+    method: 'GET',
+  };
+}
+
+function getAndAddPackageHistories(packages) {
+  const historyQueries = packages.map((item) => {
+    const request = createGetPackageHistoryRequest(item.id);
+    return query(request);
+  });
```

Comment:
I think this will cause long loading times when we have a lot of package histories. I personally think it is better to do lazy loading. We can just grab the package history if it is not stored in redux yet.